testgen/generators/base.py

```diff
@@ -20,7 +20,39 @@ def __init__(self, icu_version, run_limit=None):
     def process_test_data(self):
         pass
 
+    def generateTestHashValues(self, testdata):
+        # For each test item, copy it. Omit 'label' from that copy.
+        # Create the string representation of that copy with json.dumps()
+        # Then make  a hex hash value for that string.
+        # Add it to that item.
+
+        try:
+            all_tests =  testdata['tests']
+        except BaseException as error:
+            logging.error('# generateTestHashValues: %s does not have "tests": %s',
+                          error, testdata.keys())
+            return None
+
+        for test in all_tests:
+            try:
+                test_no_label = test.copy()
+            except BaseException as error:
+                logging.error('error: %s, Item with no label found here: %s, %s' ,
+                              error, testdata['test_type'], test)
+                continue
+            del test_no_label['label']
+            test_no_label_string = json.dumps(test_no_label)
```

Comment:
Suggestion: to make this more stable, please remove empty fields and sort them:

- https://stackoverflow.com/a/60124334/1407170
- https://stackoverflow.com/a/23820416/1407170

Something similar can be done in JSON.

Reply:
Thanks for helping address these last week. New commits fix this.